lib/charms/operator_libs_linux/v0/grub.py

```diff
@@ -161,13 +161,24 @@ def _save_config(path: Path, config: Dict[str, str], header: str = CONFIG_HEADER
     if path.exists():
         logger.debug("GRUB config %s already exist and it will overwritten", path)
 
-    context = [f"{key}={shlex.quote(value)}" for key, value in config.items()]
     with open(path, "w", encoding="UTF-8") as file:
-        file.writelines([header, *context])
+        file.write(f"{header}{os.linesep}")
+        for key, value in config.items():
+            file.write(f"{key}={shlex.quote(value)}{os.linesep}")
```

Comment:
```suggestion
            file.write(f"{key}={shlex.quote(value)}\n")
```
But yeah, this new code structure is definitely clearer than what it was before.